src/header.rs

```diff
@@ -324,6 +328,160 @@ impl NiftiHeader {
+    /// if `qform_code` has a valid value, otherwise return a "base" transformation, constructed
+    /// from the declared shape and zooms.
+    ///
+    /// If `sform_code` and `qform_code` both have valid values, the 'sform' affine transformation
+    /// is prioritized.
+    pub fn get_affine(&self) -> Affine4 {
+        if self.sform_code != 0 {
+            self.get_sform_affine()
+        } else if self.qform_code != 0 {
+            self.get_qform_affine()
+        } else {
+            self.get_base_affine()
+        }
+    }
+
+    /// Retrieve affine transformation from 'sform' fields.
+    fn get_sform_affine(&self) -> Affine4 {
+        Affine4::new(
+            self.srow_x[0], self.srow_x[1], self.srow_x[2], self.srow_x[3],
+            self.srow_y[0], self.srow_y[1], self.srow_y[2], self.srow_y[3],
+            self.srow_z[0], self.srow_z[1], self.srow_z[2], self.srow_z[3],
+            0.0, 0.0, 0.0, 1.0,
+        )
+    }
+
+    /// Retrieve affine transformation from qform-related fields.
+    fn get_qform_affine(&self) -> Affine4 {
+        if self.pixdim[1] < 0.0 || self.pixdim[2] < 0.0 || self.pixdim[3] < 0.0 {
+            panic!("All spacings (pixdim) should be positive");
+        }
+        if self.pixdim[0].abs() != 1.0 {
+            panic!("qfac (pixdim[0]) should be 1 or -1");
+        }
+
+        let quaternion = self.get_qform_quaternion();
+        let r = quaternion_to_affine(quaternion);
+        let s = Affine3::from_diagonal(&Vector3::new(
+            self.pixdim[1],
+            self.pixdim[2],
+            self.pixdim[3] * self.pixdim[0],
+        ));
+        let m = r * s;
+        Affine4::new(
+            m[0] as f32, m[3] as f32, m[6] as f32, self.quatern_x,
+            m[1] as f32, m[4] as f32, m[7] as f32, self.quatern_y,
+            m[2] as f32, m[5] as f32, m[8] as f32, self.quatern_z,
+            0.0, 0.0, 0.0, 1.0,
+        )
+    }
+
+    /// Retrieve affine transformation implied by shape and zooms.
+    ///
+    /// Note that we get the translations from the center of the image.
+    fn get_base_affine(&self) -> Affine4 {
+        let d = self.dim[0] as usize;
+        shape_zoom_affine(&self.dim[1..d + 1], &self.pixdim[1..d + 1])
+    }
+
+    /// Compute quaternion from b, c, d of quaternion.
+    ///
+    /// Fills a value by assuming this is a unit quaternion.
+    fn get_qform_quaternion(&self) -> Quaternion<f32> {
+        fill_positive(Vector3::new(self.quatern_b, self.quatern_c, self.quatern_d))
+    }
+
+    /// Set affine transformation.
+    ///
+    /// Will set both affine transformations to avoid interoperability problems:
+    ///
+    /// * 'sform' from unmodified `affine`, with `sform_code` set to `AlignedAnat`.
+    /// * 'qform' from a quaternion built from `affine`. However, the 'qform' won't be used by most
+    /// nifti readers because the `qform_code` will be set to 'Unknown'.
```

Comment:
If by `Unknown` you mean the `XForm` variant, I guess we can do this:

```suggestion
    /// nifti readers because the `qform_code` will be set to `Unknown`.
```